machine_learning/dbscan/dbscan.ipynb

```diff
@@ -0,0 +1,318 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "plt.plot(X[:,0], X[:,1],'ro')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The implementation is inspired from the original DBScan algorithm as given in \n",
+    "<a href = \"https://en.wikipedia.org/wiki/DBSCAN\">DBSCAN Wikipedia</a>\n",
+    "\n",
+    "### Abstract Algorithm\n",
+    "The DBSCAN algorithm can be abstracted into the following steps:\n",
+    "\n",
+    "- Find the points in the $ε$ (eps) neighborhood of every point, and identify the core points with more than <b>min_pts</b> neighbors.\n",
+    "- Find the connected components of core points on the neighbor graph, ignoring all non-core points.\n",
+    "- Assign each non-core point to a nearby cluster if the cluster is an $ε$ (eps) neighbor, otherwise assign it to noise.\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Preparing the points\n",
+    "Initially we label all the points in the dataset as __undefined__ .\n",
+    "\n",
+    "__points__ is our database of all points in the dataset."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "points = { (point[0],point[1]):{'label':'undefined'} for point in X }\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Helper functions"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def distFunc(Q, P):\n",
+    "    '''\n",
+    "        Calculates the Euclidean distance\n",
+    "        between pointd P and Q\n",
+    "    '''\n",
+    "    a = pow((Q[0] - P[0]),2)\n",
+    "    b = pow((Q[1] - P[1]),2)\n",
+    "    return pow((a+b),0.5)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def rangeQuery(DB,Q,eps):\n",
```

Comment:
__find_neighbors()__ instead of __rangeQuery()__ ?

Reply:
Done.